engine/debug/debug_menu.asm

```diff
@@ -72,27 +72,31 @@ ELSE
 	ret
 ENDC
 
-TestBattle:
+TestBattle: ; unreferenced (non-debug)
 .loop
 	call GBPalNormal
 
-	; Don't mess around
-	; with obedience.
+	; Don't mess around with obedience.
 	ld a, 1 << BIT_EARTHBADGE
 	ld [wObtainedBadges], a
 
 	ld hl, wFlags_D733
 	set BIT_TEST_BATTLE, [hl]
 
+	; The bag items are set here in pokeyellow's equivalent routine.
```

Comment:
What are "The bag items"? What is "pokeyellow's equivalent routine"? Why use `EXP_ALL` as an example, is it significant? What's the speculation about battle difficulty for?

Reply:
> What label is "The bag items"? What file+label is "pokeyellow's equivalent routine"?

The label is `wNumBagItems` and pokeyellow's `TestBattle` adds some items to the bag after setting `BIT_TEST_BATTLE`.

> Why use EXP_ALL as an example, is it significant?

Yes, defeating the Rhydon seems to always yield experience as if `EXP_ALL` were in the bag.

> What's the speculation about battle difficulty for?

You're more likely to lose the test battle than not, which makes it harder to notice the `EXP_ALL` issue.

Reply:
Okay, so is this accurate: In the debug ROM, without modifications or RAM-editing cheat codes, you entered the debug TestBattle and observed that exp gain at the end of the battle was awarded as if Exp All was present?

If so, how about this:
```
; wNumBagItems and wBagItems are not initialized here,
; and their garbage values happen to act as if EXP_ALL
; is in the bag at the end of the TestBattle.
; pokeyellow fixes this by initializing them with a
; list of items.
```
(Specify the label of those items if/when it's renamed from `Data_feded`. ) (I think pretty many pokeyellow-only labels are still unidentified but have clear meanings, if you feel like PRing those.)

Reply:
> Okay, so is this accurate

It is, yes.

> if you feel like PRing those

My goal is to use the final version of this PR as base for pokeyellow so that we can identify the Yellow-specific debug code and data more easily.